### PR TITLE
Allow choosing min CPU architecture for master VM on gce

### PR DESCRIPTION
**Pull request by shyamjvs**
To help with https://github.com/kubernetes/kubernetes/issues/55777

/cc @porridge 
/assign @wojtek-t 

/kind bug
/priority critical-urgent
/sig scalability

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/container-linux/master-helper.sh [hunk 1]
```diff
@@ -79,6 +79,11 @@ function create-master-instance-internal() {
     preemptible_master="--preemptible --maintenance-policy TERMINATE"
   fi
 
+  local min_cpu_selection=""
+  if [[ -n "${MASTER_MIN_CPU_ARCHITECTURE:-}" ]]; then
+    min_cpu_selection="--min-cpu-selection=${MASTER_MIN_CPU_ARCHITECTURE}"
```
Comment (porridge) [inline cluster/gce/container-linux/master-helper.sh]: The example I saw used `--min-cpu-platform` not `--min-cpu-selection`?

Comment (shyamjvs) [reply cluster/gce/container-linux/master-helper.sh]: Good catch.. fixed it.

File: cluster/gce/config-default.sh [hunk 2]
```diff
@@ -30,6 +30,7 @@ REGIONAL_KUBE_ADDONS=${REGIONAL_KUBE_ADDONS:-true}
 NODE_SIZE=${NODE_SIZE:-n1-standard-2}
 NUM_NODES=${NUM_NODES:-3}
 MASTER_SIZE=${MASTER_SIZE:-n1-standard-$(get-master-size)}
+MASTER_MIN_CPU_ARCHITECTURE=${MASTER_MIN_CPU_ARCHITECTURE:-} # To allow choosing better architectures. Also helps w/ issue #55777.
```
Comment (wojtek-t) [inline cluster/gce/config-default.sh]: I vote for removing the "Also helps w/ issue #55777." part of the comment.
Same below.

Comment (shyamjvs) [reply cluster/gce/config-default.sh]: Yeah.. that sounds reasonable ;)
Done.

**Comment by porridge** [comment]
@wojtek-t can you approve for milestone too?

**Comment by shyamjvs** [comment]
Oops... got merged before I fixed a bug in this PR. This is not WAI due to
some bash pain with spaces in the argument 'Intel Broadwell' for e.g.

**Comment by shyamjvs** [comment]
I'll fix it in a follow-up PR.

**Pull request merged**